backend/app/database/supabase/supabase_results.py

```diff
@@ -0,0 +1,36 @@
+from dataclasses import dataclass
+
+from postgrest import APIResponse
+
+from app.database import CommandResult, QueryResult
+
+
+@dataclass(frozen=True)
+class SupabaseDeleteQueryResult(CommandResult):
+    supabase_delete_query_result: APIResponse
+
+    @property
+    def succeeded(self) -> bool:
```

Comment:
The `succeeded` thing I suggested was if you're NOT using exceptions. Then you've got ~ a bespoke Either/Result monad like https://github.com/dry-python/returns?tab=readme-ov-file#result-container.